### PR TITLE
feat(vue-app): function watchQuery

### PR DESCRIPTION
**Pull request by clarkdo**
## Types of changes

- [ ] Bug fix (a non-breaking change which fixes an issue)
- [ ] New feature (a non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Description

Be merged after 2.9.2
## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

**Comment by pimlie** [comment]
I fear the watchQuery function indeed very much 😆

**Comment by clarkdo** [comment]
？

**Comment by pimlie** [comment]
See the PR title

**Comment by clarkdo** [comment]
Opps😱

File: packages/vue-app/template/client.js [hunk 1]
```diff
@@ -279,6 +279,7 @@ async function render(to, from, next) {
   // Get route's matched components
   const matches = []
   const Components = getMatchedComponents(to, matches)
+  const instances = getMatchedComponentsInstances(to)
```
Comment (pimlie) [inline packages/vue-app/template/client.js]: Could this be moved to the watchQuery `else if` (eg using memoization)?

The need for `instances` is for such a specific use-case only that we shouldnt always spent those cpu cycles retrieving them when there is a reasonable chance we dont need them

Comment (clarkdo) [reply packages/vue-app/template/client.js]: Done.

**Pull request merged**